sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -976,6 +976,60 @@ public void Main(Loader loader)
             context.WalkWithInstructions(3);
         }
 
+        [TestMethod]
+        [TestCategory("Symbolic execution")]
+        public void ExplodedGraph_TupleExpressionsWithDiscardDeconstruct()
+        {
+            const string testInput = @"
+using System.Collections.Generic;
+
+namespace Tests.Diagnostics.CSharp8
+{
+    public class Loader
+    {
+        public void Deconstruct(out string projectInstance, out List<string> diagnostics)
+        {
+            projectInstance = "";
+            diagnostics = new List<string>();
+        }
+    }
+
+    public class Test
+    {
+        public void Main(Loader loader)
+        {
+            var (projectInstance, _) = loader;
+        }
+    }
+}";
+            var context = new ExplodedGraphContext(TestHelper.Compile(testInput));
```

Comment:
Note that for simplicity you could replace the full test input by only:
```
            const string testInput = "var (projectInstance, _) = loader;";
            var context = new ExplodedGraphContext(testInput);
```
as there is no semantic check for the compilation

Reply:
Nice! It simplifies the code quite a bit.